static/js/components/UserMenu.js

```diff
@@ -9,12 +9,13 @@ import type { User } from "../flow/authTypes"
 
 type Props = {
   /* This is here for future use when we have custom profile avatars */
-  currentUser: User
+  currentUser: User,
+  useScreenOverlay: boolean
 }
 
-const UserMenu = ({ currentUser }: Props) => {
+const UserMenu = ({ currentUser, useScreenOverlay }: Props) => {
```

Comment:
There is a lot of repetition here. Instead of evaluating the same condition over and over, and returning the same result over and over, you can just calculate the attributes you need before rendering. Could look something like this:

```jsx
const overlayMenuContainerProps = {
  className: "user-menu dropdown"
}
const overlayUListProps = {
  className: "dropdown-menu"
}
const overlayListItemProps = {
  className: "authenticated-menu"
  "data-toggle": "collapse",
  "data-target": "#nav"
}
const desktopListItemProps = {
  className: "dropdown-item"
}

type MenuChildProps = {
  container?: Object,
  ul?: Object,
  li?: Object
}

const UserMenu = ({ currentUser, useScreenOverlay }: Props) => {
  let menuChildProps: MenuChildProps
  menuChildProps = useScreenOverlay ? ({
    container: overlayMenuContainerProps,
    ul: overlayUListProps,
    li: overlayListItemProps
  }) : ({
    li: desktopListItemProps
  })
  return (
    <div { ...(menuChildProps.container || {})}>
      //...
      <ul
        aria-labelledby="dropdownMenuButton"
        { ...(menuChildProps.ul || {})}
      >
        <li { ...(menuChildProps.li || {})}>
          //...
        </li>
      </ul>
    </div>
  )
}
```

When you're evaluating the same boolean condition many times over, and you often have the same result after you evaluate it, that's a code smell and you should look for ways to refactor.

Please refactor `AnonymousMenu.js` in a similar way as well